### PR TITLE
Improved readability of Cucumber tests

### PR DESCRIPTION
**Pull request by AlexaVB**

### CODE REVIEW

File: features/step_definitions/common_steps.rb [hunk 1]
```diff
@@ -114,9 +114,9 @@ def group_common(index)
   expect(hint_group.text).to have_content hint
 end
 
-Then(/^I am taken to step (\d+) ([^\"]*)$/) do |step, url|
+Then(/^I am taken to step (\d+) - ([^\"]*)$/) do |step, title|
   expect(common_page.content.p.text).to have_content step
-  expect(current_path).to eq "/questions/#{url}"
+  expect(common_page.content.h1.text).to eq title
```
Comment (garytaylor) [inline features/step_definitions/common_steps.rb]: Using h1 doesnt seem right - we use page objects to abstract boring technical definitions like this - can the h1 not be called something meaningful from a user's perspective ?

File: features/step_definitions/step_eight_steps.rb [hunk 2]
```diff
@@ -3,45 +3,36 @@ def group_eight(index)
...
-And(/^I select working tax credit from income list$/) do
-  expect(group_eight(0).block_label[2].text).to eq 'Working Tax Credit'
-  group_eight(0).block_label[2].click
-  common_page.continue_button.click
-end
-
-When(/^I select no income$/) do
-  group_eight(0).input[12].click
+When(/^I submit the form with no income checked$/) do
+  group_eight(0).no_income.click
```
Comment (garytaylor) [inline features/step_definitions/step_eight_steps.rb]: I wondered before what this group_{number}({another_number}) means ? it does not read well at all imo.  Can this be defined in a different way to make it more readable ?

File: features/step_definitions/step_eight_steps.rb [hunk 3]
```diff
@@ -53,8 +44,8 @@ def group_eight(index)
 end
 
 Then(/^I should see an income list for myself and my partner:$/) do |incomes|
-  expect(group_eight(0).h3.text).to eq 'Your income'
-  expect(group_eight(1).h3.text).to eq 'Your partner\'s income'
+  expect(group_eight(0)).to have_your_income
```
Comment (garytaylor) [inline features/step_definitions/step_eight_steps.rb]: What does have_your_income mean ?  I cant see this defined anywhere ?

Comment (AlexaVB) [reply features/step_definitions/step_eight_steps.rb]: That'll be in the page object 'element :your_income, 'h3', text: 'Your income''

File: features/step_definitions/step_eight_steps.rb (hunk 2 above)
Comment (AlexaVB) [reply features/step_definitions/step_eight_steps.rb]: I'm sure there is, would appreciate your help on this but will continue to refactor on a new branch

File: features/step_definitions/common_steps.rb (hunk 1 above)
Comment (AlexaVB) [reply features/step_definitions/common_steps.rb]: I know and I would appreciate your help on this but the user doesn't see the steps, only we do. I don't want to add ambiguity by calling the h1 something else

**Pull request merged**